### DISCUSSION TITLE
cellClassName & headerClassname v8

### DISCUSSION THREAD

**Discussion by laurent512** [TanStack/table#4100] [Q&A] [answered]
Hi, 

Is there some cellClassName & headerClassname properties in the column definition in V8 ?

My goal is to define specific className for specific cells ? 

Current way of doing it is `cell: (props) => <div classsName="...">{props.getValue()}</div>`,

Thank you !

**Comment by jordanjustice** [comment, marked as answer]
@laurent512 I was struggling to find the same thing for a few days, but stumbled across this question: https://github.com/TanStack/table/discussions/4097

With the clue there, I'm now defining a `cellClassName` and `headerClassName` in the column def's `meta` object.

```tsx
const defaultColumns: ColumnDef<Submission>[] = [
  {
    id: 'actions',
    header: '',
    size: 20,
    minSize: 20,
    maxSize: 20,
    enableResizing: false,
    meta: {
      cellClassName: 'w-4'
    },
    cell: () => <div className="w-5 h-5 bg-red-600 rounded-full" />
  }
]

...
// The `td` element in the table now has this for grabbing the classes (note the optional chaining)
<td
  key={cell.id}
  className={classNames(
    'px-4 py-2.5 text-sm font-sans text-blue-gray-900 break-words whitespace-nowrap',
    cell.column.columnDef.meta?.cellClassName
)}>
  {flexRender(cell.column.columnDef.cell, cell.getContext())}
</td>
...
```

**Reply by marceloverdijk** [reply]
Does it work well with TypeScript warning/errors or did you declared the `cellClassName` via [declaration merging](https://tanstack.com/table/v8/docs/api/core/column-def#meta)?
I'm trying to do something similar but I'm struggling with TypeScript warning/errors: https://github.com/TanStack/table/discussions/4157

**Comment by tannerlinsley** [comment]
Column meta is the answer ;)
